test/sequential/test-net-connect-emfile.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const net = require('net');
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+
+console.log('Test to check the error message format during EMFILE error');
+
+if (process.platform === 'win32') {
+  console.log('Skipping test, no RLIMIT_NOFILE on Windows.');
+  return;
+}
+
+var totalErrorCount = 0;
+
+const reactToError = function() {
+  totalErrorCount += 1;
+  if (totalErrorCount === 2) {
+    console.log('Test OK');
+    process.exit(0);
+  }
+};
+
+// Server listens
+net.createServer().listen(common.PORT);
+
+// Exhaust the file descriptors
+try {
+  while (true) {
+    fs.openSync(__filename, 'r');
+  }
+} catch (err) {
+  assert(err.code === 'EMFILE' || err.code === 'ENFILE');
+}
+
+// Now connect to the server. We should get EMFILE error from `getaddrinfo`
+const client1 = net.connect({
+  port: common.PORT
+});
+
+client1.on('error', function(er) {
+  assert.equal(er.message, 'getaddrinfo EMFILE');
+  assert.equal(er.code, 'EMFILE');
+  assert.equal(er.syscall, 'getaddrinfo');
```

Comment:
The reason I bring this up is that getaddrinfo() doesn't necessarily create a new file descriptor; it may have cached one or use some platform-specific mechanism that doesn't involve file descriptors.